File: yemeksiparismete_mobile/lib/models/restaurant.dart
class Restaurant {
  final int id;
  final String? ownerId;
  final String name;
  final String? logo;
  final String? logoUrl;
  final String? bannerUrl;
  final String category;
  final String address;
  final String description;
  final double rating;

  Restaurant({
    required this.id,
    this.ownerId,
    required this.name,
    this.logo,
    this.logoUrl,
    this.bannerUrl,
    required this.category,
    required this.address,
    required this.description,
    required this.rating,
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: json['id'] ?? 0,
      ownerId: json['ownerId'],
      name: json['name'] ?? '',
      logo: json['logo'],
      logoUrl: json['logoUrl'],
      bannerUrl: json['bannerUrl'],
      category: json['category'] ?? '',
      address: json['address'] ?? '',
      description: json['description'] ?? '',
      rating: (json['rating'] ?? 4.5).toDouble(),
    );
  }
}

File: yemeksiparismete.Server/Models/Restaurant.cs
namespace yemeksiparismete.Server.Models
{
    public class Restaurant
    {
        public int Id { get; set; }
        
        // Restoranın sahibi olan kullanıcının ID'si (İlişkisel Veritabanı mantığı)
        public string? OwnerId { get; set; } 
        public string? Logo { get; set; } // Compatibility with frontend mock data        
        public string Name { get; set; } = string.Empty;
        
        public string LogoUrl { get; set; } = string.Empty;
        public string BannerUrl { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        
        public string Address { get; set; } = string.Empty;
        
        public string Description { get; set; } = string.Empty;

        public int DistrictId { get; set; }
        
        public double Rating { get; set; } = 4.5;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}